internal/builders/bazel/build.sh

```diff
@@ -16,8 +16,7 @@
 
 set -euo pipefail
 
-# TODO(Issue #2331): switch copy to binaries to a temp dir
-mkdir binaries
+mkdir bazel_builder_binaries_to_upload_to_gh_7bc972367cb286b7f36ab4457f06e369
```

Comment:
nit: add a comment why there's a long string in the name, ie avoid collision.
You may as well make it a variable and use it in your code:
```shell
my_name="blabla"
...
"./${my_name}
```